### DISCUSSION TITLE
ApplicationMessageReceived in server

### DISCUSSION THREAD

**Discussion by mdjavedakhtar** [dotnet/MQTTnet#1220] [Q&A] [answered]
i am trying to use ApplicationMessageReceived inside server
server starting up and client can subscribe

            mqttServer.UseApplicationMessageReceivedHandler(e =>{}

but i am not getting the messeges in server.

can anyone show me a sample code for same ??

**Comment by mdjavedakhtar** [comment]
mqttServer.ApplicationMessageReceivedHandler = new MqttApplicationMessageReceivedHandlerDelegate(OnAppMessage);
this worked
OnAppMessage has to be async

**Comment by SeppPenner** [comment, marked as answer]
The `ApplicationMessageReceivedHandler` in the server works the same as in the client, but has the clientId as additional field: https://github.com/chkr1011/MQTTnet/wiki/Server#consuming-messages, https://github.com/chkr1011/MQTTnet/wiki/Client#consuming-messages.

Your example code with the delegate should work, as you found out, with `async` methods.
